sdk/core/azure-core/src/http/curl/curl_transport.cpp

```diff
@@ -45,6 +45,33 @@ CURLcode CurlTransport::Perform(Context& context, Request& request)
     return settingUp;
   }
 
+  if (request.GetMethod() == HttpMethod::Get)
+  {
+    settingUp = curl_easy_setopt(m_pCurl, CURLOPT_HTTPGET, 1L);
+  }
+  else if (request.GetMethod() == HttpMethod::Put)
+  {
+    settingUp = curl_easy_setopt(m_pCurl, CURLOPT_PUT, 1L);
+  }
+  else
+  {
+    settingUp
+        = curl_easy_setopt(m_pCurl, CURLOPT_CUSTOMREQUEST, HttpMethodToString(request.GetMethod()).data());
```

Comment:
I have removed this while merging to my branch.
Is it a good idea to use this `CURLOPT_CURSOMREQUEST` ?  
Documentation says it changes the value in the request but doesn't change how libcurl makes the request. So if we set up libcurl to do GET and then use this opt to set PUT, libcurl will still perform GET

Reply:
I left out POST here, we don't have to use CURLOPT_CUSTOMREQUEST for GET, PUT and POST.

Without CURLOPT_CUSTOMREQUEST, how are you going to do HEAD, PATCH, DELETE?